src/rpa_gw.F

```diff
@@ -1451,6 +1498,38 @@ SUBROUTINE compute_QP_energies(vec_Sigma_c_gw, count_ev_sc_GW, gw_corr_lev_occ,
 
    END SUBROUTINE compute_QP_energies
 
+! **************************************************************************************************
+!> \brief ...
+!> \param vec_gw_energ ...
+!> \param z_value ...
+!> \param m_value ...
+! **************************************************************************************************
+   SUBROUTINE check_NaN(vec_gw_energ, z_value, m_value)
```

Comment:
What do you think about the following suggestion:
The routine checks only a single array (ALLOCATABLE attribute is not necessary here) and sets NaNs to a user-given value.

Reply:
Yes, good point, we can certainly do, code is updated.